screeps-game-api/src/constants.rs

```diff
@@ -830,3 +925,57 @@ impl TryFrom<Value> for ResourceType {
         })
     }
 }
+
+impl TryFrom<String> for ResourceType {
+    type Error = String;
+
+    fn try_from(s: String) -> Result<Self, Self::Error> {
```

Comment:
See comment on `to_string`: I'd prefer to use integers to translate the various types across the rust<->js boundary whenever possible.

Not against having this implemented, but I think it'd be good to at least mention the str->int translations available on the JS side in a doc-comment here, and to warn about not using this if not necessary.